src/utils/video.ts

```diff
@@ -10,3 +10,41 @@ export const formatVideoViewsAndDate = (
   const formattedViews = views !== null && (fullViews ? formatNumber(views) : formatNumberShort(views))
   return formattedViews ? `${formattedViews} views • ${formattedDate}` : formattedDate
 }
+
+export type VideoMetadata = {
+  mimeType: string
+  sizeInBytes: number
+  height: number
+  width: number
+  duration: number
+}
+export const getVideoMetadata = async (
+  file?: File
```

Comment:
Any reason to allow an `undefined` file here?

Reply:
Yes, the file we get from the input is of type `File | undefined` so this way it's easier to type/use

Reply:
But that's handling the input. The controller should check input's value and based on that either call this function or not. It doesn't make sense to call this function with `undefined`

Reply:
True, that makes sense